biogrid/libs/biogrid-simulator/src/lib/community/ruralarea.ts

```diff
@@ -6,51 +6,50 @@ import { Building } from '../building';
  */
 export class RuralArea implements Town {
 
-  private town: Building[];
+  private buildings: Building[] = [];
+  private randomIdNumbers : Number[] = [];
 
   /**
-   * @param {Building[]} town A list of buildings which make up a town.
+   * @param {Building[]} buildings A list of buildings which make up a town.
    */
-  constructor(town: Building[]) {
-    this.town = town;
-    this.setCurrentBuildingIds();
+  constructor(buildings: Building[]) {
+    for (let i=0; i<buildings.length; i++) {
+      this.addBuilding(buildings[i]);
+    }
   }
 
-  getTown() : Building[] {
-    return this.town;
+  getBuildings() : Building[] {
+    return this.buildings;
   }
 
   /**
-   * This method gets a building in a town by its name identifier.
-   * @param {Building} building The building we're looking for.
+   * This method gets a building in a town by its id number.
+   * @param {number} Id The building Id we're looking for.
    * @return {Building} Returns the building if found in the list, null if not.
    */
   getBuildingById(Id: number) {
-    for (let i=0; i<this.town.length; i++) {
-      if (this.town[i].getBuildingId() === Id) {
-        return this.town[i];
+    for (let i=0; i<this.buildings.length; i++) {
+      if (this.buildings[i].getBuildingId() === Id) {
+        return this.buildings[i];
       }
     }
     return null;
   }
 
   /**
-   * This method adds a building to our town.
+   * This method adds a building to our town and assigns it a random Id.
    * @param {Building} newBuilding The building to be added. 
    */
   addBuilding(newBuilding: Building) : Building {
-    newBuilding.setBuildingId(this.town.length);
-    this.town.push(newBuilding);
+    let randomId = Math.floor((Math.random() * 1000));
+    while (this.randomIdNumbers.includes(randomId)) {
+      randomId = Math.floor((Math.random() * 1000));
+    }
+    this.randomIdNumbers.push(randomId);
```

Comment:
currently there is no way to remove a building, but if that function is ever implemented, you'll have to remember to also remove that building's ID from that list. If you want to avoid having to remember that, you could create the list of randomIdNumbers within this function by looping through the list of buildings, instead of manually updating it here. You can do this either with a for loop or with an Array.map() function (which lets you take an array a, and apply some sort of transformation to each value, and return an array b where every value of a had your transformation applied). For example:

```
// for every building in this.buildings, return the building ID in a new array that only holds these ids
const randomIds = this.buildings.map(building => building.getBuildingId());
let randomId = <your code>
while (randomIds.includes(randomId) ..... <your code>
```

Let me know if that makes sense!

Reply:
That does! Thank you for simplifying that!